action/protocol/staking/protocol.go

```diff
@@ -475,7 +475,7 @@ func (p *Protocol) ActiveCandidates(ctx context.Context, sr protocol.StateReader
 	list := c.AllCandidates()
 	cand := make(CandidateList, 0, len(list))
 	for i := range list {
-		if ok && featureCtx.AddContractStakingVotes {
+		if ok && p.contractStakingIndexer != nil && featureCtx.AddContractStakingVotes {
```

Comment:
1. need to use `MustGetFeatureCtx` to make sure we can access the `AddContractStakingVotes` flag here
2. move if before for loop
```
if p.contractStakingIndexer != nil && protocol.MustGetFeatureCtx(ctx).AddContractStakingVotes {
   for i := range list
}
```

Reply:
During service startup , `ActiveCandidates` is called at `nodeinfo/manager.go:104`, and the provided context does not contain the required feature. Using `MustGetFeatureCtx`  will lead to a panic.

Reply:
is it possible to add a feature context there?

Reply:
BTW, file a small separate PR for this